### PR TITLE
WEBDEV-787_Alphabetise-root-list

### PR DESCRIPTION
**Pull request by Christine-horrocks**

### CODE REVIEW

File: app/serializers/page_serializer/groups_show_page_serializer.rb [hunk 1]
```diff
@@ -8,6 +8,7 @@ class GroupsShowPageSerializer < PageSerializer::BasePageSerializer
     def initialize(request: nil, group: nil, data_alternates: nil)
       @group = group
       @layings = @group.try(:layingBodyHasLaying)
+      @layings = SortHelper.sort_by_reverse(collection: @layings, attributes: [:date, :laidThingName]) if @layings
```
[the remark this answers is not in the document]
Comment (Christine-horrocks) [reply app/serializers/page_serializer/groups_show_page_serializer.rb]: I changed this to meet with hounds suggestion, but when I checked it again it told me I needed to change it to its current implementation. It cannot be appeased. 😠

File: app/controllers/concerns/sort_helper.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module SortHelper
```
Comment (mattrayner) [inline app/controllers/concerns/sort_helper.rb]: Can you update this to implement a sort_by method

Then also implement the sort_by_reverse, but it's implementation should be something like:

```ruby
def self.sort_by_reverse(collection: nil, attributes: nil, prepend_rejected: false)
  self.sort_by(
    collection: collection,
    attributes: attributes,
    prepend_rejected: prepend_rejected
  ).reverse
end
```

File: app/serializers/page_serializer/groups_show_page_serializer.rb [hunk 3]
```diff
@@ -54,24 +55,24 @@ def section_literals
...
     end
 
     def card_heading(laying)
       ComponentSerializer::HeadingComponentSerializer.new(content: laying.laid_thing.try(:laidThingName), size: 2, link: statutory_instrument_path(laying.laid_thing.graph_id)).to_h
     end
 
-    def card_paragraph(laying)
-      ComponentSerializer::ParagraphComponentSerializer.new(content: [{ content: "Laying date: #{l(laying.date)}" }]).to_h
+    def card_list(laying)
+      ComponentSerializer::ListDescriptionComponentSerializer.new(items: [{ 'term': { 'content': 'laid-thing.laid-date' }, 'description': [{ 'content': l(laying.date) }] }]).to_h
```
Comment (mattrayner) [inline app/serializers/page_serializer/groups_show_page_serializer.rb]: this line would be clearer if it was split across multiple lines

File: app/serializers/page_serializer/groups_show_page_serializer.rb [hunk 4]
```diff
@@ -54,24 +55,24 @@ def section_literals
...
+        items << { 'term': { 'content': 'Name' }, 'description': [{ 'content': @group.groupName }] } if @group.try(:groupName)
         items << { 'term': { 'content': 'Start Date' }, 'description': [{ 'content': l(@group.start_date) }] } if @group.try(:groupStartDate)
         items << { 'term': { 'content': 'End Date' }, 'description': [{ 'content': l(@group.end_date) }] } if @group.try(:groupEndDate)
       end.compact
     end
 
     def objects
       @layings.map do |laying|
-        ComponentSerializer::CardComponentSerializer.new(name: 'card__generic', data: { card_type: 'small', heading: card_heading(laying), paragraph: card_paragraph(laying) }).to_h
+        ComponentSerializer::CardComponentSerializer.new(name: 'card__generic', data: { card_type: 'small', heading: card_heading(laying), list_description: card_list(laying) }).to_h
```
Comment (mattrayner) [inline app/serializers/page_serializer/groups_show_page_serializer.rb]: this line would be clearer if it was split across multiple lines

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -51,50 +51,50 @@
...
-        index = object.index('"')
-
-        object[0..index] = new_object if index
-
-        new_line = "#{subject}> <#{predicate}> \"#{object}"
-        config.filter_sensitive_data(new_line) { line }
-      end
-    end
-  end
+  # # Dynamically filter n-triple data
```
Comment (mattrayner) [inline spec/spec_helper.rb]: Should this still be commented out? Maybe this is a larger discussion about whether we need this any more

File: app/controllers/concerns/sort_helper.rb (hunk 2 above)
Comment (Christine-horrocks) [reply app/controllers/concerns/sort_helper.rb]: 🎉

File: app/serializers/page_serializer/groups_show_page_serializer.rb (hunk 3 above)
Comment (Christine-horrocks) [reply app/serializers/page_serializer/groups_show_page_serializer.rb]: 🎉

File: app/serializers/page_serializer/groups_show_page_serializer.rb (hunk 4 above)
Comment (Christine-horrocks) [reply app/serializers/page_serializer/groups_show_page_serializer.rb]: 🎉

File: spec/spec_helper.rb (hunk 5 above)
Comment (Christine-horrocks) [reply spec/spec_helper.rb]: I have uncommented it, but I think it might be reasonable to have a conversation about if it is still needed.

**Pull request merged**